components/server/src/iam/iam-session-app.spec.ts

```diff
@@ -231,6 +237,58 @@ class TestIamSessionApp {
         expect(this.teamDbMock.memberships.has(BUILTIN_INSTLLATION_ADMIN_USER_ID)).to.be.false;
         expect(this.teamDbMock.memberships.has("id-new-user")).to.be.true;
     }
+
+    @test public async testSessionRequest_updates_existing_user() {
+        const payload: OIDCCreateSessionPayload = { ...this.payload };
+        payload.claims.sub = this.knownSubjectID; // `userServiceMock.findUserForLogin` will match this value
+
+        this.knownUser.identities = [
+            {
+                authId: payload.claims.sub,
+                authProviderId: payload.claims.aud,
+                authName: "Test User",
+                primaryEmail: "random-user@any.org",
+            },
+        ];
+
+        let newEmail: string | undefined;
+        this.userServiceMock.updateUserIdentity = async (user, updatedIdentity) => {
+            newEmail = updatedIdentity.primaryEmail;
+        };
+
+        const result = await request(this.app.create())
+            .post("/session")
+            .set("Content-Type", "application/json")
+            .send(JSON.stringify(payload));
+
```

Comment:
It feels this whole test is quite brittle. We're mostly testing that once you hit `/session` with some payload, the handler invokes the `userServiceMock.updateUserIdentity` method which is in fact a stub. So here, we're just checking that it invokes our stub, rather than it actually performs the behaviour we want.

Is it possible to instead use the real service and verify the record got updated in the DB?

Reply:
It is, if we make it a db-test.

Reply:
Yes, this way it makes assumptions about the internal paths of the code.